### PR TITLE
Add .npmrc file

### LINKED ISSUE #1670: npmrc file missing for test project
The Secure Supply Chain Analysis build step is warning of a missing npmrc file:

```
##[warning]tests/testproject/nodeapp/package.json - CFS0001: Missing sibling .npmrc file. (https://aka.ms/cfs/npm)
```

Related to https://github.com/dotnet/aspire/issues/1419

Let's get these SCCA warnings cleaned up for Preview 1. These show up in [VMR](https://github.com/dotnet/dotnet) builds, causing warnings which makes it difficult to identify when another new warnings pop up.

### PR DESCRIPTION
**Pull request by eerhardt**
Resolves a Security Supply Chain Violation

I copied this file and approach from https://github.com/dotnet/aspnetcore/blob/main/.npmrc.

Fix #1670
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1722)

### CODE REVIEW

**Review by mthalman — APPROVED** [review]

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+registry=https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public-npm/npm/registry/
+always-auth=true
```
Comment (sebastienros) [inline .npmrc:2]: I think this line is specific to aspnet core, and not even sure it's required anymore.
Had to deal with it during the holidays. If you look at all the other dotnet repos they don't have it. It's supposed to allow azdo to authenticate to the feeds (all the upstream ones) using the tokens configured in azdo. But these are public and shouldn't require auth. That's what I understood.

**Review by sebastienros — APPROVED** [review]

File: .npmrc (hunk 1 above)
Comment (sebastienros) [reply .npmrc:2]: @wtgodbe

Comment (dougbu) [reply .npmrc:2]: @sebastienros not exactly. other repos don't do this b/c they don't use `npm` or `yarn`. and, auth is for updating the feed when an authorized person like you grabs packages from the upstream feed and updates lock files

Comment (sebastienros) [reply .npmrc:2]: Javier told me 2 weeks ago and I already had forgotten.

So I believe we don't need that here. There is nothing that requires the feed to be updated (packages.lock file), right @eerhardt ?

Comment (eerhardt) [reply .npmrc:2]: I'm just following the docs from the error: https://aka.ms/cfs/npm. See `Mitigation Steps` section.

Comment (eerhardt) [reply .npmrc:2]: I merged what I had to unblock the supply chain validation. We can tweak this as we go.

**Pull request merged**